---
title: "QU39 Zooplankton Time series"
---


# 0. Prepare environment

Load necessary packages and import functions.

```{r}
# Data import
library(googledrive)
library(readxl)
library(phyloseq)
source("Code/DownloadDatasets.R")


# All data format modifications and filtering and plotting
library(tidyverse)
'%!in%' <- function(x,y)!('%in%'(x,y))
source("Code/Functions.R")
library(zoo)

# Statistical tools
library(vegan)

# Specific plotting tools
library(plotly)
library(ggpubr)
library(paletteer)
library(viridis)
library(cowplot)
#library(ggnewscale)
source("Code/scales_and_themes.R")

# Make sure tidyverse is the latest loaded package
detach("package:tidyverse", unload=TRUE)
library(tidyverse)
```



The flowing code will download the needed data sets from Google Drive server and put them in the Data_import directory. Only has to be run once after cloning this directory.

```{r eval=FALSE, include=FALSE}
# Confirm authorisation with googledrive
drive_auth()

# Update datasets.
# These functions refere to code specifyed under Code/Import_datasets.R.
update_DNA_datasets()
update_Contributed_data()

```

# ### DATA PREP. ###

# 1. eDNA Data components

The COI Sequencing data are downloaded in the compact format with a separate ASV, Taxonomy, and Sample Data table. The data tables are read separately and then merged into a tidy data tibble, resembling the data format of most zooplankton count data sets.

```{r}
# Import downloaded rds (compact r data) files
ASV_COI <- readRDS("Data_import/COI/ASV_COI.rds")
Samptab_COI <- readRDS("Data_import/COI/Samptab_COI.rds")
Taxtab_COI <- readRDS("Data_import/COI/Taxtab_COI.rds")

# Combine the three components to a tidy data tibble
df_COI <-
  full_join(Taxtab_COI, ASV_COI, by = "ASV") %>%
  pivot_longer(!1:12, names_to = "Library_ID", values_to = "Abundance") %>%
  full_join(Samptab_COI, by = "Library_ID")
```


## List Sampling and Extraction Contributions:
```{r}
personel <- Samptab_COI %>%
  filter(year(Sample_date) == 2017) %>% 
  group_by(Sample_date) %>% 
  summarize(Sample_technician = unique(Sample_technician),
            Extraction_staff = unique(Extraction_staff)) %>%
  pivot_longer(2:3, names_to = "Task", values_to = "technician") %>% 
  separate(technician, sep = ",", into = c("A", "B", "C", "D")) %>% 
  pivot_longer(3:6, values_to = "tecnichian") %>%
  filter(is.na(tecnichian) == FALSE) %>% 
  select(!name)

# Sampling staff  
personel %>% 
  filter(Task == "Sample_technician") %>%
  group_by(tecnichian) %>% 
  summarise(Days_Sampled = n()) %>% 
  arrange(-Days_Sampled)

# Extraction staff
personel %>% 
  filter(Task == "Extraction_staff") %>%
  group_by(tecnichian) %>% 
  summarise(Samples_Extracted = n()) %>% 
  arrange(-Samples_Extracted)
```

List contributions for QU39 microbial:

```{r}


personel <- read_excel("./Data_import/2024-02-09_170138_HakaiData_microbial.xlsx", sheet = "Hakai Data") %>%
  filter(year(date) %in% c("2015", "2016", "2017", "2018", "2019")) %>% 
  filter(microbial_sample_type == "DNA") %>% 
  group_by(date) %>% 
  summarize(lab_technician = unique(lab_technician),
            extracting_staff = unique(extracting_staff)) %>%
  pivot_longer(2:3, names_to = "Task", values_to = "technician") %>% 
  separate(technician, sep = ",", into = c("A", "B", "C", "D")) %>% 
  pivot_longer(3:6, values_to = "tecnichian") %>%
  filter(is.na(tecnichian) == FALSE) %>% 
  select(!name)

# Sampling staff  
personel %>% 
  filter(Task == "lab_technician") %>%
  group_by(tecnichian) %>% 
  summarise(Days_Sampled = n()) %>% 
  arrange(-Days_Sampled)

# Extraction staff
personel %>% 
  filter(Task == "extracting_staff") %>%
  group_by(tecnichian) %>% 
  summarise(Samples_Extracted = n()*5) %>% 
  arrange(-Samples_Extracted)






```


This is the contributed data set from Catherinas 18S sequencing project. The data is downloaded as a Phyloseq object, and melted into a data frame to resemble the data above.

```{r}
df_18S_ts <- readRDS("Data_import/Contributed18S/ps_18S.rds") %>% psmelt()
```



```{r}
# Import downloaded rds (compact r data) files
ASV_12S <- readRDS("Data_import/12S/ASV_12S.rds")
Samptab_12S <- readRDS("Data_import/12S/Samptab_12S.rds")
Taxtab_12S <- readRDS("Data_import/12S/Taxtab_12S.rds")


# Combine the three components to a tidy data tibble
df_12S <-
  full_join(Taxtab_12S, ASV_12S, by = "ASV") %>% 
  pivot_longer(!1:11, names_to = "Library_ID", values_to = "Abundance") %>% 
  full_join(Samptab_12S, by = "Library_ID")
```


## Calculate eDNA Index and clean up datasetds

In this section the different data formats of the 2017 QU39 time series will be mutated into similar data shapes and transformed in such a way that the data types can be compared. To keep filtering parameters equal for all data types, a set of functions are first defined, that are then executed for all data sets. Comparisons are done both at species and at genus level.


```{r}
# Filter samples in the dataset
eDNA_COI <- df_COI %>%
  filter(Project_name == "QU39",
         Sample_type == "eDNA",
         phylum %in% metazoa)

# Species level index
eDNA_COI_index_species <- eDNA_COI %>% 
  mutate(Taxa = lca.taxon,
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex(Library_ID, Hakai_ID)

# Genus level ind
eDNA_COI_index_genus <- eDNA_COI %>% 
  mutate(Taxa = genus,
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex(Library_ID, Hakai_ID)


```


```{r}
# Clean up datasets
COI_genus <- eDNA_COI_index_genus %>% 
  filter(Taxa %in% getGenusList(.)) %>% 
  mutate(Depth = as.numeric(Depth),
         Depth = ifelse(Depth>200, 250, Depth),
         Date = as_date(Date)) %>%
  filter(is.na(Depth)==F) %>%
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>100)


COI_species <- eDNA_COI_index_species %>% 
  filter(Taxa %in% getSpeciesList(.)) %>% 
  mutate(Depth = as.numeric(Depth),
         Depth = ifelse(Depth>200, 250, Depth),
         Date = as_date(Date)) %>%
  filter(is.na(Depth)==F) %>%
  
  # Exclude libraries with too few reads
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>500)

COI_species

saveRDS(COI_species, "Data_output/COI_species.rds")
```


# 2. Environmental Data components

This is the list of Hakai_ID's we have COI data for: 
```{r}
samples_COI <- COI_species %>% 
  pull(Hakai_ID) %>% 
  unique()
```

Extract Hakai Microbial metadata for samples
```{r}
QMIC <- read_excel("./Data_import/2024-02-09_170138_HakaiData_microbial.xlsx", sheet = "Hakai Data") %>% 
  filter(hakai_id %in% samples_COI) %>% 
  arrange(event_pk)

niskin <- QMIC %>% 
  mutate(NISKIN = paste(event_pk, line_out_depth, sep = "_")) %>% 
  select(hakai_id,date, line_out_depth, NISKIN ) %>% 
  mutate(date = as_date(date))
niskin
```

Chlorophyll bottle data
```{r}
CHL <- read_excel("./Data_import/2023-12-21_112549_HakaiData_chlorophyll_15-20.xlsx", sheet = "Hakai Data") %>% 
  mutate(NISKIN = paste(event_pk, line_out_depth, sep = "_")) %>% 
  filter(NISKIN %in% niskin$NISKIN) %>% 
  #select(NISKIN, filter_type, chla, phaeo) %>%
  group_by(NISKIN, filter_type) %>% 
  summarise(chla = mean(chla),
            phaeo = mean(phaeo)) %>%
  pivot_wider(names_from = filter_type, values_from = c(chla, phaeo))
CHL
```

Nutrient bottle data
```{r}
NUT <- read_excel("./Data_import/2023-12-21_113016_HakaiData_nutrients.xlsx", sheet = "Hakai Data") %>% 
  mutate(NISKIN = paste(event_pk, line_out_depth, sep = "_")) %>% 
  filter(NISKIN %in% niskin$NISKIN) %>% 
  select(NISKIN, no2_no3_um, po4, sio2) %>% 
  group_by(NISKIN) %>% 
  summarise(no2_no3_um = mean(no2_no3_um),
            po4 = mean(po4),
            sio2 = mean(po4))
NUT
```

CTD Bottle data
```{r}
# Identify the correct CTD drops (Can be improved)
drops <- read_excel("./Data_import/CTD.xlsx", sheet = "Drops") %>% 
  mutate(date = as_date(`Start time`)) %>% 
  filter(date %in% niskin$date,
         `Target depth (m)` > 240) %>% 
  group_by(date) %>% 
  summarise(CastPK = min(`Cast PK`))

# Load and modify CTD data, filter by drop.
CTD <- read_excel("./Data_import/CTD.xlsx", sheet = "Data") %>% 
  transmute(CastPK = `Cast PK`,
            Distance_from_station = `Distance from station`,
            Station,
            Time = `Measurement time`,
            Date = as_date(Time), 
            Depth = `Depth (m)`,
            Conductivity = `Conductivity (mS/cm)`,
            Temperature = `Temperature (deg C)`,
            Pressure = `Pressure (dbar)`,
            PAR = `PAR (umol m-2 s-1)`,
            Chlorophyll = `Fluorometry Chlorophyll (ug/L)`,
            Turbidity = `Turbidity (FTU)`,
            O2 = `Dissolved O2 (mL/L)`,
            Salinity = `Salinity (PSU)`) %>% 
  filter(CastPK %in% drops$CastPK)

# Function for extracting depth data.
closest <- function(dat, m) {
  dat %>% 
  group_by(Date) %>% 
  slice(which.min(abs(Depth - m))) %>% 
  mutate(Depth = m)
}

# Extract data for 0, 5, 30, 100 and 260 m.
ctd_bottle <- bind_rows(
  closest(CTD, 0),
  closest(CTD, 5),
  closest(CTD, 30),
  closest(CTD, 100),
  closest(CTD, 260)) %>%
  select(date = Date, line_out_depth = Depth, Temperature, Salinity, PAR, Turbidity, O2)

ctd_bottle

```

Merge all datasets to bottle file:
```{r}
bottle_file <- niskin %>% 
  left_join(CHL, by = "NISKIN") %>% 
  left_join(NUT, by = "NISKIN") %>% 
  left_join(ctd_bottle, by = c("date", "line_out_depth"))

saveRDS(bottle_file, "Data_output/bottle_file.rds")
```

# ### PLOTS & STATS ###

# 1. Diagnostic plots

## Average relative abundace
Species level:
```{r}
COI_species %>%
  filter(Abundance > 0) %>%
  group_by(Taxa) %>% 
  mutate(maxRA = median(RA)) %>%
  ungroup() %>% 
  mutate(Taxa = fct_reorder(Taxa, -maxRA)) %>% 
  ggplot() +
  geom_boxplot(aes(Taxa, RA)) +
  theme_minimal_hgrid(font_size = 8) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,
                                   face = "italic"),
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey')) +
  scale_y_continuous(trans='log10')

Abundant_Species <- COI_species %>% 
  group_by(Taxa) %>%
  mutate(maxRA = median(RA)) %>%
  #filter(maxRA >= 0.01) %>% # 15 species
  #filter(max(RA) >0.14) %>% # 15 species
  filter(max(RA) >0.24) %>% # 10 Species
  pull(Taxa) %>%  unique
Abundant_Species

```

## Library coverage
```{r}
COI_species %>% 
  group_by(Library_ID, Depth, Date) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  mutate(Depth = factor(Depth, levels = c(
    250, 100, 30, 5, 0
  ))) %>% 
  ggplot() +
  geom_raster(aes(Date, Depth)) +
  geom_vline(xintercept = datebreaks, color = "#F5F5F5", size = 0.2, alpha = 0.8) +
  geom_vline(xintercept = yearbreaks, color = "grey", size = 0.2, alpha = 0.8) +
  scale_x_date(date_labels="%b %y", breaks  = datebreaks) +
  #scale_fill_gradient2(high = '#4daf4a') +
  theme_minimal_vgrid(font_size = 8) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/4)


```


## eDNA profiles
```{r}
# Make species interpolation:
Interp_Species <- COI_species %>%
  filter(Taxa %in%Abundant_Species) %>%
  group_by(Taxa) %>%
  group_modify(~interpolate_rollmean(.x, x1 = Date, x2 = Depth, y = RA_Index,
                                     # res 
                                     x1res = 1, x2res = 1, k = 14,
                                     dim = 1))

# Plot dataset
Interp_Species %>%
  mutate(Depth = factor(Depth, levels = c(
    250, 100, 30, 5, 0
  ))) %>% 
  ggplot() +
  geom_raster(aes(Date, Depth, fill = RA_Index)) +
  geom_vline(xintercept = datebreaks, color = "#F5F5F5", size = 0.2, alpha = 0.8) +
  geom_vline(xintercept = yearbreaks, color = "grey", size = 0.2, alpha = 0.8) +
  facet_wrap("Taxa", ncol = 2) +
  #scale_y_reverse() +
  scale_x_date(date_labels="%b %y", breaks  = datebreaks) +
  #scale_fill_viridis(option = "H") +
  scale_fill_gradient2(high = 'green',
                       low = "white") +
  theme_minimal_vgrid(font_size = 8) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        #legend.position="none",,
        aspect.ratio = 1/4)

ggsave("./Figure_output/COI_Interpolation.pdf")

```

Plot lines instead

```{r}
Interp_Species %>%
  filter(Taxa %in% Abundant_Species) %>% 
  mutate(Depth = factor(Depth)) %>%
  ggplot() +
  geom_line(aes(Date, RA_Index, color = Depth)) +
  facet_wrap("Taxa", scales = "free", ncol = 2) +
  scale_x_date(date_labels="%b %y", breaks  = datebreaks) +
  theme_minimal_vgrid(font_size = 10) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        #strip.text = element_text(face = "italic"),
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        #legend.position="none",,
        aspect.ratio = 1/4) +
  scale_color_manual(values = depth_colors)
  
  ggsave("./Figure_output/COI_Interpolation_lines.pdf")
  

```



# 2. Environmental Data

## Bottle data:

```{r}
Bottle_File <- bottle_file %>% 
  dplyr::transmute(Hakai_ID = hakai_id,
                   Date = date,
                   Depth = line_out_depth,
                   depth = factor(line_out_depth, levels = c("260", "100",
                                                             "30", "5", "0")),
                   Temperature = Temperature,
                   Salinity = Salinity,
                   PAR = PAR,
                   Turbidity = Turbidity,
                   O2 = O2,
                   `chlA (>20um)` = chla_20um,
                   `chlA (>3um)` = chla_3um,
                   `chlA (>0.7um)` = `chla_GF/F`,
                   `phaeo (>20um)` = phaeo_20um,
                   `phaeo (>20um)` = phaeo_3um,
                   `phaeo (>20um)` = `phaeo_GF/F`,
                   `NO2 & NO3` =  no2_no3_um,
                   PO4 = po4
                   ) %>% 
  pivot_longer(5:15, names_to = "Variable", values_to = "value") %>% 
  dplyr::filter(is.na(value) == F) %>% 
  dplyr::mutate(Variable = factor(Variable, levels = c(
    "Temperature", "Salinity",
    "chlA (>20um)", "chlA (>3um)", "chlA (>0.7um)",
    "phaeo (>20um)", "phaeo (>3um)", "phaeo (>0.7um)",
    "NO2 & NO3", "PO4",
    "O2", "PAR", "Turbidity")))
Bottle_File



Bottle_File %>%
  filter() %>% 
  ggplot() +
  geom_line(aes(Date, value, color = depth)) +
  facet_wrap("Variable", scales = "free", ncol = 2) +
  scale_x_date(date_labels="%b %y", breaks  = datebreaks) +
  theme_minimal_vgrid(font_size = 10) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        #strip.text = element_text(face = "italic"),
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        #legend.position="none",,
        aspect.ratio = 1/4) +
  scale_color_manual(values = depth_colors)

ggsave("./Figure_output/Bottle_Data.pdf")

```


# 3. Community analysis:

## CCA

```{r}
Abundant_Species <- COI_species %>% 
  group_by(Taxa) %>%
  mutate(maxRA = median(RA)) %>%
  filter(max(RA) >0.1) %>% # 32 species
  pull(Taxa) %>%  unique
Abundant_Species

comm <- COI_species %>% 
  filter(Taxa %in% Abundant_Species) %>% 
  dplyr::group_by(Hakai_ID, Taxa) %>% 
  summarise(Index = mean(RA_Index)) %>%
  pivot_wider(names_from = "Taxa", values_from = "Index") %>% 
  arrange(Hakai_ID) %>% 
  column_to_rownames("Hakai_ID")

env <- bottle_file %>% 
  arrange(hakai_id) %>%
  mutate(Depth = factor(line_out_depth),
         Month = month(date)) %>%
  column_to_rownames("hakai_id")
```

Model filt with stepwise reduction
```{r}
# Model all environmental parameters
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month +
             chla_20um + chla_3um +`chla_GF/F` + `chla_Bulk GF/F` +
             phaeo_20um + phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F` +
             no2_no3_um + po4,
           data = env,
           na.action = na.omit)
vif.cca(mod)

# Remove po4
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month +
             chla_20um + chla_3um +`chla_GF/F` + `chla_Bulk GF/F` +
             phaeo_20um + phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F` +
             no2_no3_um,
           data = env,
           na.action = na.omit)
vif.cca(mod)

# Remove chl_20
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month + 
             chla_3um +`chla_GF/F` + `chla_Bulk GF/F` +
             phaeo_20um + phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F` +
             no2_no3_um,
           data = env,
           na.action = na.omit)
vif.cca(mod)

# Remove phaeo_20
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month + 
             chla_3um +`chla_GF/F` + `chla_Bulk GF/F` +
             phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F` +
             no2_no3_um,
           data = env,
           na.action = na.omit)
vif.cca(mod)

# chla_GF/F
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month + 
             chla_3um + `chla_Bulk GF/F` +
             phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F` +
             no2_no3_um,
           data = env,
           na.action = na.omit)
vif.cca(mod)

# Remove `chla_Bulk GF/F` 
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month + 
             chla_3um +
             phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F` +
             no2_no3_um,
           data = env,
           na.action = na.omit)
vif.cca(mod)

#remove no2_no3_um
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month + 
             chla_3um +
             phaeo_3um + `phaeo_GF/F` + `phaeo_Bulk GF/F`,
           data = env,
           na.action = na.omit)
vif.cca(mod)
plot(mod)

#remove phaeo_3um
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             Depth + Month + 
             chla_3um +
             `phaeo_GF/F` + `phaeo_Bulk GF/F`,
           data = env,
           na.action = na.omit)
vif.cca(mod)
plot(mod)



env
```

Alt 2 only include selected from beginning:


```{r}
# Model all relevant environmental parameters
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             chla_20um + chla_3um +`chla_GF/F` + 
             no2_no3_um + po4,
           data = env,
           na.action = na.omit)
vif.cca(mod)
plot(mod)

# Remove PO4
mod <- cca(comm ~
             Temperature + Salinity + PAR + Turbidity + O2 +
             chla_20um + chla_3um +`chla_GF/F` +
             no2_no3_um,
           data = env,
           na.action = na.omit)
vif.cca(mod)
plot(mod)


```



```{r}
#extracting the data as data frame; env data
veg_1 = as.data.frame(mod$CCA$biplot) %>% 
  rownames_to_column("env")


#extracting the data; genusv
veg_2 = as.data.frame(mod$CCA$v) %>% 
  rownames_to_column("Taxa")

plot <-  ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  #geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue") +
  ggrepel::geom_text_repel(data = veg_2,
                           aes(x = CCA1, y = CCA2, label = Taxa),
                           nudge_y = -0.05) +

  geom_segment(data = veg_1,
               aes(x = 0, y = 0, xend = CCA1, yend = CCA2),
               arrow = arrow(length = unit(0.25, "cm"))) +
  ggrepel::geom_text_repel(data = veg_1,
                           aes(x = CCA1, y = CCA2, label = env),
                           nudge_y = -0.05,
                           color = "blue",
                           size = 5) +
  theme_bw() +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
plot

```




## NMDS

```{r}
seasonal_colors = c("#053061", "#3288BD", "#66C2A5", "#7FBC41",
                    "#A6D96A", "#FEE08B", "#FDAE61", "#F46D43",
                    "#D53E4F", "#9E0142", "#67001F", "#40004B")




## WRAPPER
plotNMDS <- function(x) {
  

mat <- COI_species %>%
  filter(Depth == x) %>% 
  dplyr::select(Library_ID, Taxa, RA_Index) %>% 
  pivot_wider(names_from = Taxa, values_from = RA_Index) %>% 
  column_to_rownames("Library_ID") %>% 
  as.matrix()
  
  mat
  
mod <- metaMDS(mat, trymax = 20)

# Extract NMDS data
{  Samples <- 
    scores(mod)$sites %>% 
    as.data.frame() %>%
    rownames_to_column("Library_ID") %>% 
    mutate(Layer = "Samples")
  
  Taxa <-
    scores(mod)$species %>% 
    as.data.frame() %>%
    rownames_to_column("Taxa") %>% 
    mutate(Layer = "Taxa")
  
  NMDS_res <- bind_rows(Samples, Taxa) %>% 
    pivot_wider(names_from = Layer,
              values_from = c(NMDS1, NMDS2))
}

# Merge with metadata
COI_species %>% 
  dplyr::select(Date, Depth, Library_ID) %>%
  mutate(Month = format(as.Date(Date), format="%m"),
         Depth = factor(Depth)) %>%
  left_join(NMDS_res, by = "Library_ID") %>% 
  ggplot(aes(NMDS1_Samples, NMDS2_Samples)) +
  geom_text(aes(NMDS1_Taxa, NMDS2_Taxa, label = Taxa),
            size = 3, colour = "grey", fontface = "italic") +
  geom_point(aes(color = Month, shape = Depth), size = 3) +
  scale_color_manual(values = seasonal_colors)


}

plotNMDS(0)
plotNMDS(5)
plotNMDS(30)
plotNMDS(100)
plotNMDS(250)


```












